WalletWasabi/Extensions/NBitcoinExtensions.cs

```diff
@@ -263,5 +263,19 @@ public static void SortByAmount(this TxOutList list)
 		{
 			list.Sort((x, y) => x.Value.CompareTo(y.Value));
 		}
+
+		public static void SortByAmount(this TxInList list, List<Coin> coins)
+		{
+			var map = new Dictionary<TxIn, Coin>();
+			foreach (var coin in coins)
+			{
+				Predicate<TxIn> txInFinder = (TxIn txIn) =>
```

Comment:
You don't need to declare the `txInFinder` for each coin, declare it outside the loop.